### ISSUE TITLE
Broken parameter links in some of the stdlib documentation

### ISSUE THREAD

**Issue by RossSmyth** [rust-lang/rust#121106]
1. Go to https://doc.rust-lang.org/std/simd/type.u8x1.html#method.len
2. Click on the usize return parameter
3. The link goes here: https://doc.rust-lang.org/primitive.usize.html

This happens with other parameters. [But if you go to the SIMD page through the prelude the link is not broken](https://doc.rust-lang.org/std/simd/prelude/struct.Simd.html#method.len). Nor is it broken on the portable simd docs. There are other things broken, such as slices on the SIMD page as well. It is also on other SIMD types' documentation pages. I don't know the exact pattern though.

### I expected to see this happen: 

The link would be to this page: https://doc.rust-lang.org/std/primitive.usize.html

### Instead, this happened: 

The link is to this page: https://doc.rust-lang.org/primitive.usize.html

Basically it seems to not be inserting the "std" directory in the link.

**Comment by gurry** [comment]
Regressed in one of these two commits: db3e2bacb 74cf505e365

**Comment by gurry** [comment]
Ignore the above comment. It was a wrong bisection.

The regression most likely occurred in PR #116471. I say "most likely" because it was impossible to bisect around this part of the git log because builds kept failing with ICE's. 

Anyway, the proximate cause of the bug is that in type alias docs links to primitives are being rendered relative to the target type not the alias. For example for `u8x1` the link is rendered as `../../primitive.<prim>.html` instead of `../primitive.<prim>.html` because the target type `Simd` is two levels deep from the root even though `u8x1` is just one level deep.

This problem seems limited only type aliases.

Internally this is how the rendering of the link works:
- `write_shared()` generates the primitive path as just `primitive.<prim>.html` in the HTML fragment held in the JSONP format.
- `window.register_type_impls` in `main.js` then prepends this string with the value of `window.rootPath` which seems to be relative to the target type (i.e. `../../' for our above example of `u8x`)

The end result is that we get a link relative to the type not the alias.

**Comment by gurry** [comment]
The fix could either be in `main.js` wherein we could compute the correct path to prepend depending on the path of the alias instead of just using `window.rootPath`. Or it could be in `write_shared()` /`primitive_link_fragment()` methods whereby we could generate the path  `std/primitive.<prim>.html` instead of `primitive.<prim>.html` for the primitive which will ensure that just prepending `window.rootPath` in front of it will work.

What are your thoughts @notriddle since you where the original author? This is the first time I'm ventured into rustdoc so any pointers will be helpful.
